lib/compiler.js

```diff
@@ -101,6 +101,43 @@ Compiler.prototype.visitText = function (node) {
 }
 
 /**
+ * Handle a "Literal" include such as an SVG or HTML file
+ */
+Compiler.prototype.visitLiteral = function (node) {
+  const generateWidgetStr = `
+    function generateWidget(contents) {
```

Comment:
my general take here is that you should do it one of two ways:
1. `generateWidget` as a dynamically-generated function which takes no params and has `escapedSource` injected into its definition (`this.contents = ${escapedSource};`)
2. `generateWidget` as a static function that takes contents as a param (like here), but is only defined once and only if a template actually needs it. you can see a similar example with `__objToAttrs` getting added as a bit of 'runtime code' at the top of the generated template here: https://github.com/tdumitrescu/virtual-jade/blob/71ae2cbf/lib/compiler.js#L54-L61

my preference is probably for option 2, since otherwise one could imagine a template with a lot of SVGs getting bloated quickly with repeated definitions of this function.